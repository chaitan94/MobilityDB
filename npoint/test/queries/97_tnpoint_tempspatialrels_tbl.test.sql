-------------------------------------------------------------------------------
-- Geometry rel tnpoint
-------------------------------------------------------------------------------

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tcontains(ST_SetSRID(t1.g, 5676), t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tcovers(ST_SetSRID(t1.g, 5676), t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tcoveredby(ST_SetSRID(t1.g, 5676), t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tdisjoint(ST_SetSRID(t1.g, 5676), t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tequals(ST_SetSRID(t1.g, 5676), t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tintersects(ST_SetSRID(t1.g, 5676), t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE ttouches(ST_SetSRID(t1.g, 5676), t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE twithin(ST_SetSRID(t1.g, 5676), t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tdwithin(ST_SetSRID(t1.g, 5676), t2.temp, 0.01) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE trelate(ST_SetSRID(t1.g, 5676), t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE trelate(ST_SetSRID(t1.g, 5676), t2.temp, 'T*****FF*') IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;

-------------------------------------------------------------------------------
-- Geometry rel tnpoint
-------------------------------------------------------------------------------

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tcontains(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tcovers(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tcoveredby(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tdisjoint(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tequals(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tintersects(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE ttouches(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE twithin(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tdwithin(t1.np, t2.temp, 0.01) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE trelate(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE trelate(t1.np, t2.temp, 'T*****FF*') IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;

-------------------------------------------------------------------------------
-- tnpoint rel <Type>
-------------------------------------------------------------------------------

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tcontains(t1.temp, ST_SetSRID(t2.g, 5676)) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tcovers(t1.temp, ST_SetSRID(t2.g, 5676)) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tcoveredby(t1.temp, ST_SetSRID(t2.g, 5676)) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tdisjoint(t1.temp, ST_SetSRID(t2.g, 5676)) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tequals(t1.temp, ST_SetSRID(t2.g, 5676)) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tintersects(t1.temp, ST_SetSRID(t2.g, 5676)) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE ttouches(t1.temp, ST_SetSRID(t2.g, 5676)) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE twithin(t1.temp, ST_SetSRID(t2.g, 5676)) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tdwithin(t1.temp, ST_SetSRID(t2.g, 5676), 0.01) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE trelate(t1.temp, ST_SetSRID(t2.g, 5676)) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE trelate(t1.temp, ST_SetSRID(t2.g, 5676), 'T*****FF*') IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tcontains(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tcovers(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tcoveredby(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tdisjoint(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tequals(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tintersects(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE ttouches(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE twithin(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tdwithin(t1.temp, t2.np, 0.01) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE trelate(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE trelate(t1.temp, t2.np, 'T*****FF*') IS NOT NULL AND t1.k < 5 AND t2.k < 5;

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tdisjoint(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tequals(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tintersects(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tdwithin(t1.temp, t2.temp, 0.01) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE trelate(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE trelate(t1.temp, t2.temp, 'T*****FF*') IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;

-------------------------------------------------------------------------------
