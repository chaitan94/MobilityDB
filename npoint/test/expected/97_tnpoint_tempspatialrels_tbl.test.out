SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tcontains(t1.g, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tcovers(t1.g, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tcoveredby(t1.g, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tdisjoint(t1.g, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tequals(t1.g, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tintersects(t1.g, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE ttouches(t1.g, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE twithin(t1.g, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE tdwithin(t1.g, t2.temp, 0.01) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE trelate(t1.g, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_geometry t1, tbl_tnpoint t2 WHERE trelate(t1.g, t2.temp, 'T*****FF*') IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
   380
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tcontains(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tcovers(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tcoveredby(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tdisjoint(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tequals(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tintersects(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE ttouches(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE twithin(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE tdwithin(t1.np, t2.temp, 0.01) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE trelate(t1.np, t2.temp) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_npoint t1, tbl_tnpoint t2 WHERE trelate(t1.np, t2.temp, 'T*****FF*') IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    80
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tcontains(t1.temp, t2.g) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    76
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tcovers(t1.temp, t2.g) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    76
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tcoveredby(t1.temp, t2.g) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    76
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tdisjoint(t1.temp, t2.g) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    76
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tequals(t1.temp, t2.g) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    76
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tintersects(t1.temp, t2.g) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    76
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE ttouches(t1.temp, t2.g) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    76
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE twithin(t1.temp, t2.g) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    76
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE tdwithin(t1.temp, t2.g, 0.01) IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    76
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE trelate(t1.temp, t2.g) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    12
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_geometry t2 WHERE trelate(t1.temp, t2.g, 'T*****FF*') IS NOT NULL AND t1.k < 5 AND t2.k%25 < 5;
 count 
-------
    76
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tcontains(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tcovers(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tcoveredby(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tdisjoint(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tequals(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tintersects(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE ttouches(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE twithin(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE tdwithin(t1.temp, t2.np, 0.01) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE trelate(t1.temp, t2.np) IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_npoint t2 WHERE trelate(t1.temp, t2.np, 'T*****FF*') IS NOT NULL AND t1.k < 5 AND t2.k < 5;
 count 
-------
    16
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tcontains(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tcovers(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tcoveredby(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tdisjoint(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tequals(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tintersects(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE ttouches(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE twithin(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE tdwithin(t1.temp, t2.temp, 0.01) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE trelate(t1.temp, t2.temp) IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

SELECT count(*) FROM tbl_tnpoint t1, tbl_tnpoint t2 WHERE trelate(t1.temp, t2.temp, 'T*****FF*') IS NOT NULL AND t1.k%25 < 5 AND t2.k%25 < 5;
 count 
-------
    20
(1 row)

